Gems/Streamer/StreamerProfiler/Code/Source/StreamerProfilerSystemComponent.cpp

```diff
@@ -0,0 +1,776 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <StreamerProfilerSystemComponent.h>
+
+#include <AzCore/Casting/lossy_cast.h>
+#include <AzCore/Casting/numeric_cast.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/EditContextConstants.inl>
+#include <AzCore/IO/IStreamer.h>
+#include <AzCore/IO/Streamer/FileRequest.h>
+
+#if defined(IMGUI_ENABLED)
+#include <imgui/imgui.h>
+#endif
+
+namespace Streamer
+{
+    StreamerProfilerSystemComponent::GraphStore::GraphStore()
+    {
+        memset(m_values.data(), 0, sizeof(float) * GraphStoreElementCount);
+    }
+
+    StreamerProfilerSystemComponent::GraphStore::GraphStore(float minValue, float maxValue)
+        : m_minValue(minValue)
+        , m_maxValue(maxValue)
+    {
+        memset(m_values.data(), 0, sizeof(float) * GraphStoreElementCount);
```

Comment:
suggest that you add an assert to make sure minValue is never larger than maxValue

Reply:
Added